internal/writer/write.go

```diff
@@ -11,13 +11,19 @@ import (
 
 type WriteOptions struct {
 	Save      bool
+	Name      string
 	Directory string
 }
 
 func Write(syscalls []uint32, functionSymbol string, opts WriteOptions) error {
 	var errOut error
 	if opts.Save {
-		fileName := archiver.Convert(functionSymbol)
+		var fileName string
+		if opts.Name != "" {
+			fileName = opts.Name
+		} else {
+			fileName = archiver.Convert(functionSymbol)
+		}
```

Comment:
This might be simplify no?

```suggestion
		fileName := archiver.Convert(functionSymbol)
		if opts.Name != "" {
			fileName = opts.Name
		}
```

Or this

```suggestion
		fileName := opts.Name
		if fileName == "" {
			fileName = archiver.Convert(functionSymbol)
		}
```